packages/protocol/test/tokenomics/proofReward.test.ts

```diff
@@ -0,0 +1,250 @@
+import { expect } from "chai";
+import { SimpleChannel } from "channel-ts";
+import { ethers } from "ethers";
+import { TaikoL1 } from "../../typechain";
+import { TestTkoToken } from "../../typechain/TestTkoToken";
+import { randEle } from "../utils/array";
+import blockListener from "../utils/blockListener";
+import Proposer from "../utils/proposer";
+import Prover from "../utils/prover";
+import { createAndSeedWallets } from "../utils/seed";
+import { commitProposeProveAndVerify } from "../utils/verify";
+import { initTokenomicsFixture } from "./utils";
+
+describe("tokenomics: proofReward", function () {
+    let taikoL1: TaikoL1;
+    let l2Provider: ethers.providers.JsonRpcProvider;
+    let l1Signer: any;
+    let proposerSigner: any;
+    let proverSigner: any;
+    let genesisHeight: number;
+    let tkoTokenL1: TestTkoToken;
+    let interval: any;
+    let chan: SimpleChannel<number>;
+
+    beforeEach(async () => {
+        ({
+            taikoL1,
+            l2Provider,
+            l1Signer,
+            proposerSigner,
+            proverSigner,
+            genesisHeight,
+            tkoTokenL1,
+            interval,
+        } = await initTokenomicsFixture());
+        chan = new SimpleChannel<number>();
+    });
+
+    afterEach(() => {
+        clearInterval(interval);
+        l2Provider.off("block");
+        chan.close();
+    });
+
+    it(`proofReward is 1 wei if the prover does not hold any tkoTokens on L1`, async function () {
+        const { maxNumBlocks, commitConfirmations } = await taikoL1.getConfig();
+
+        const proposer = new Proposer(
+            taikoL1.connect(proposerSigner),
+            l2Provider,
+            commitConfirmations.toNumber(),
+            maxNumBlocks.toNumber(),
+            0,
+            proposerSigner
+        );
+
+        const prover = new Prover(taikoL1, l2Provider, proverSigner);
+        let proposed: boolean = false;
+        l2Provider.on("block", function (blockNumber: number) {
+            if (proposed) {
+                chan.close();
+                l2Provider.off("block");
+                return;
+            }
+            proposed = true;
+
+            chan.send(blockNumber);
+        });
+
+        /* eslint-disable-next-line */
+        for await (const blockNumber of chan) {
+            const proverTkoBalanceBeforeVerification =
+                await tkoTokenL1.balanceOf(await prover.getSigner().address);
+
+            await commitProposeProveAndVerify(
+                taikoL1,
+                l2Provider,
+                blockNumber,
+                proposer,
+                tkoTokenL1,
+                prover
+            );
+
+            const proverTkoBalanceAfterVerification =
+                await tkoTokenL1.balanceOf(await prover.getSigner().address);
```

Comment:
shall we change some `await prover.getSigner().address` in this file to `await prover.getSigner().getAddress()`? looks like `prover.getSigner()` dosen't return a `Promise`, but we add an `await` before it?